node/bindnode/repr.go

```diff
@@ -1029,6 +1031,36 @@ func (w *_unionAssemblerRepr) AssembleValue() datamodel.NodeAssembler {
 		valAsm := (*_unionAssembler)(w).AssembleValue()
 		valAsm = assemblerRepr(valAsm)
 		return valAsm
+	case schema.UnionRepresentation_Stringprefix:
+		// NOTE: duplicated in node.go, unionAssembler.AssembleValue
+		// consider refactoring these
+		name := w.curKey.val.String()
+		var idx int
+		var mtyp schema.Type
+		for i, member := range w.schemaType.Members() {
+			if member.Name() == name {
+				idx = i
+				mtyp = member
+				break
+			}
+		}
+		if mtyp == nil {
+			return _errorAssembler{fmt.Errorf("bindnode TODO: missing member %s in %s", name, w.schemaType.Name())}
+		}
+
+		goType := w.val.Field(idx).Type().Elem()
+		valPtr := reflect.New(goType)
+		finish := func() error {
+			unionSetMember(w.val, idx, valPtr)
+			return nil
+		}
+		return &_assembler{
+			cfg:        w.cfg,
+			schemaType: mtyp,
+			val:        valPtr.Elem(),
+			finish:     finish,
+		}
+
```

Comment:
This (and the related switch cases added to AssembleKey and Finish) doesn't belong here.  If someone is working at the representation level, and the strategy is `stringprefix`, the only thing it should accept is `AssignString`, or `AssignNode` given something that reports `Kind() == Kind_String`.